### PR TITLE
Globalise demo changes

### PR DESCRIPTION
**Pull request by BasLee**
Many small comments, we should try to find a way to divide the work.
(Larger tasks like refactoring the configuration should be done after merging.)

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -1,32 +1,43 @@
...
+      <div className="mx-auto flex w-full flex-row px-6">
         <div className="flex flex-row items-center justify-start">
-          <div className="bg-brand1-100 text-brand1-800 -mx-6 -my-3 flex flex-row items-center justify-start gap-3 py-3 pl-3">
-            <span className="uppercase">
+          <div className="bg-brand1-100 text-brand1-800 -mx-6 -my-3 flex flex-row items-center justify-start gap-3 py-3 pr-3">
+            <div className="bg-brand2-200 flex h-12 w-12 items-center justify-center">
+              <a
+                title="Homepage"
+                rel="noreferrer"
+                target="_blank"
```
Comment (BasLee) [inline src/components/Header.tsx:13]: do we want to open the home page in a new tab?

File: src/components/Header.tsx [hunk 2]
```diff
@@ -1,32 +1,43 @@
...
+            </div>
+            <span className="">
               <a
                 title="Homepage"
                 rel="noreferrer"
                 href="/"
                 className="hover:text-brand1-900 text-inherit no-underline hover:underline"
               >
-                {props.project}
+                GLOBALISE Transcriptions Viewer
```
Comment (BasLee) [inline src/components/Header.tsx:26]: should be configurable

File: src/components/Mirador/MiradorConfig.ts [hunk 3]
```diff
@@ -4,12 +4,13 @@ export const miradorConfig: MiradorConfig = {
   id: "mirador",
   window: {
     allowClose: false,
-    allowWindowSideBar: false,
+    allowWindowSideBar: true,
```
Comment (BasLee) [inline src/components/Mirador/MiradorConfig.ts:7]: should probably be configurable?

File: src/components/Projects/globalise/RenderMetadataPanel.tsx [hunk 4]
```diff
@@ -1,25 +1,59 @@
...
+            target="_blank"
+          >
+            {pageAnnotation[0].id}
+          </a>
+        </li>
+      </ul>
+    );
+  }
+
+  function renderMetadataPanelAnnotationView() {
```
Comment (BasLee) [inline src/components/Projects/globalise/RenderMetadataPanel.tsx:46]: create MetadataPanelAnnotationView component

File: src/components/Search/KeywordFacet.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      <div key={index} className="w-full max-w-[450px]">
+        <div className="font-semibold">
+          {props.searchFacetTitles[facetName] ?? facetName}
+        </div>
+        {Object.entries(facetValues)
+          //   .slice(0, visibleItems)
+          .map(([facetValueName, facetValueAmount]) => {
+            const key = `${facetName}-${facetValueName}`;
+            return (
+              // <CheckboxFacet
```
Comment (BasLee) [inline src/components/Search/KeywordFacet.tsx:34]: remove dead code

File: src/components/Search/KeywordFacet.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                          facetValueName)}
+                  </label>
+                </div>
+                <div className="text-sm text-neutral-500">
+                  {facetValueAmount}
+                </div>
+              </div>
+            );
+          })}
+        {/* {props.getKeywordFacets().length > visibleItems && (
```
Comment (BasLee) [inline src/components/Search/KeywordFacet.tsx:78]: remove dead code

File: src/components/Search/Search.tsx [hunk 7]
```diff
@@ -166,6 +185,27 @@ export const Search = (props: SearchProps) => {
     }
   }, [props.projectConfig, searchParams]);
 
+  function getTextToHighlight(data: SearchResult) {
+    const regex = new RegExp(/<em>(.*?)<\/em>/g);
+
+    const newMap = new Map<string, string[]>();
+
+    data?.results.forEach((result) => {
```
Comment (BasLee) [inline src/components/Search/Search.tsx:193]: data should never be undefined? maybe check early

File: src/components/Search/Search.tsx [hunk 8]
```diff
@@ -271,11 +332,15 @@ export const Search = (props: SearchProps) => {
       sortOrder,
     );
 
-    // const target = document.getElementsByClassName("searchContainer")[0];
```
Comment (BasLee) [inline src/components/Search/Search.tsx:274]: remove dead code

File: src/components/Search/Search.tsx [hunk 9]
```diff
@@ -299,11 +364,15 @@ export const Search = (props: SearchProps) => {
...
```
Comment (BasLee) [inline src/components/Search/Search.tsx:302]: remove dead code

File: src/components/Search/Search.tsx [hunk 10]
```diff
@@ -551,15 +570,15 @@ export const Search = (props: SearchProps) => {
...
               className="bg-brand2-100 text-brand2-700 hover:text-brand2-900 disabled:bg-brand2-50 active:bg-brand2-200 disabled:text-brand2-200 rounded px-2 py-2 text-sm no-underline"
             >
-              Nieuwe zoekopdracht
+              New search query
             </Link>
           </div>
         ) : null}
-        <div className="w-full max-w-[450px]">
+        {/* <div className="w-full max-w-[450px]">
           <SearchQueryHistory
```
Comment (BasLee) [inline src/components/Search/Search.tsx:582]: Delete or make configurable

File: src/components/Search/Search.tsx [hunk 11]
```diff
@@ -568,29 +587,41 @@ export const Search = (props: SearchProps) => {
...
             disabled={queryHistory.length === 0 ? true : false}
           />
-        </div>
+        </div> */}
         <div className="w-full max-w-[450px]">
           <Fragmenter onChange={fragmenterSelectHandler} value={fragmenter} />
         </div>
-        {renderDateFacets()}
-        {checkboxStates.size > 0 && renderKeywordFacets()}
+        {/* <div className="w-full max-w-[450px]">
```
Comment (BasLee) [inline src/components/Search/Search.tsx:594]: Delete or make configurable

File: src/components/Search/Search.tsx [hunk 12]
```diff
@@ -599,8 +630,8 @@ export const Search = (props: SearchProps) => {
           </div>
         ) : null}
         {searchResults ? (
-          <div className="border-brand1Grey-100 -mx-10 my-8 flex flex-row flex-wrap items-center justify-start gap-2 border-b px-10 pb-8">
-            <span className="text-brand1Grey-600 text-sm">Filters: </span>
+          <div className="border-brand1Grey-100 -mx-10 mb-8 flex flex-row flex-wrap items-center justify-end gap-2 border-b px-10">
+            {/* <span className="text-brand1Grey-600 text-sm">Filters: </span>
             {getKeywordFacets().map(([facetName, facetValues]) => {
```
Comment (BasLee) [inline src/components/Search/Search.tsx:635]: remove dead code

File: src/utils/matchString.ts [hunk 13]
```diff
@@ -0,0 +1,5 @@
+export function matchString(toMatchString: string, regex: RegExp) {
```
Comment (BasLee) [inline src/utils/matchString.ts:1]: method seems redundant?

File: vite.config.local.ts [hunk 14]
```diff
@@ -10,27 +10,32 @@ export default defineConfig(({ mode }) => {
   const env = loadEnv(mode, process.cwd(), "");
 
   let tailwindConfig: string;
+  let title: string;
 
   switch (env.VITE_PROJECT) {
     case "republic":
       tailwindConfig = "tailwind.config.republic.js";
+      title = "Textannoviz - Republic";
```
Comment (BasLee) [inline vite.config.local.ts:18]: move title to config file

File: deploy/nginx/nginx.conf [hunk 15]
```diff
@@ -3,9 +3,9 @@ events { }
 http {
   server {
     listen 80;
-    server_name host.docker.internal;
+    server_name localhost;
     location / {
-      proxy_pass http://frontend:3000/;
+      proxy_pass http://gltavdemo:80/;
```
[the remark this answers is not in the document]
Comment (svandaalen) [reply deploy/nginx/nginx.conf:8]: Move to project config

File: deploy/nginx/nginx.local
Comment (svandaalen) [inline deploy/nginx/nginx.local:1]: Delete

File: index.html
Comment (svandaalen) [inline index.html:1]: Make configurable in .env file

File: package.json [hunk 16]
```diff
@@ -43,7 +44,7 @@
     "@typescript-eslint/parser": "^5.53.0",
     "@vitejs/plugin-react": "^4.0.3",
     "autoprefixer": "^10.4.14",
-    "esbuild": "^0.18.17",
+    "esbuild": "^0.18.20",
```
Comment (svandaalen) [inline package.json:47]: Check why `esbuild` should always be the latest version when building

File: postcss.config.js [hunk 17]
```diff
@@ -1,5 +1,7 @@
 export default {
   plugins: {
+    "postcss-import": {},
```
Comment (svandaalen) [inline postcss.config.js:3]: Both can be removed.

File: src/Detail.tsx [hunk 18]
```diff
@@ -94,47 +94,46 @@ export const Detail = (props: DetailProps) => {
       setAnnotations(broccoli.anno);
       setViews(broccoli.views);
 
-      if (params.tier2) {
-        let annoToZoom: AnnoRepoAnnotation[];
-        if (params.tier2.includes("resolution")) {
```
Comment (svandaalen) [inline src/Detail.tsx:99]: Make zoomAnno function configurable and make annotation type configurable in project config.

File: src/app.tsx [hunk 19]
```diff
@@ -34,7 +34,7 @@ const { aggs } = await sendSearchQuery({}, "Scan", 0, 0, config);
 
 const router = createBrowserRouter(
   config.createRouter(
-    <Home project={project} config={config} />,
+    <About project={project} config={config} />,
```
[the remark this answers is not in the document]
Comment (svandaalen) [reply src/app.tsx:37]: Remove homepage in all projects and move /search to /

[the remark this answers is not in the document]
Comment (svandaalen) [reply src/app.tsx:37]: About page in other projects should be a placeholder

File: src/components/About.tsx [hunk 20]
```diff
@@ -0,0 +1,10 @@
+import { ProjectConfig } from "../model/ProjectConfig";
+
+interface AboutProps {
+  project: string;
+  config: ProjectConfig;
+}
+
+export default function About(props: AboutProps) {
+  return props.config.renderAbout();
```
[the remark this answers is not in the document]
Comment (svandaalen) [reply src/components/About.tsx:9]: Can now be merged

File: src/components/Footer.tsx [hunk 21]
```diff
@@ -17,40 +18,47 @@ type FooterProps = {
...
+  const projectConfig = useProjectStore((state) => state.projectConfig);
+
   return (
     <div className="border-brand1Grey-100 drop-shadow-top fixed bottom-0 w-full border-t bg-white text-sm text-neutral-500">
       <div className="mx-auto flex w-full flex-col justify-between lg:flex-row">
         <div className="flex w-full flex-row justify-start gap-8 lg:w-2/5">
           <button className="flex flex-row items-center gap-1 py-1 pl-10 text-neutral-500">
             <Link
-              to="/search"
+              to="/"
```
Comment (svandaalen) [inline src/components/Footer.tsx:29]: For other projects, /search should also be changed to /

File: src/components/Footer.tsx [hunk 22]
```diff
@@ -17,40 +18,47 @@ type FooterProps = {
...
               className="hover:text-brand1-600 active:text-brand1-700 text-inherit no-underline"
             >
               <MagnifyingGlassIcon className="inline h-4 w-4 fill-neutral-500" />{" "}
-              Terug naar zoekscherm
+              New search
```
Comment (svandaalen) [inline src/components/Footer.tsx:33]: See issue #18

File: src/components/Footer.tsx [hunk 23]
```diff
@@ -62,7 +70,7 @@ export const Footer = (props: FooterProps) => {
             className="hover:text-brand1-600 active:text-brand1-700 flex flex-row items-center gap-1 py-1 pr-6 text-neutral-500"
             onClick={props.showAnnotationPanelHandler}
           >
-            {props.panelShowState === true ? "Verberg info" : "Toon info"}{" "}
+            {props.panelShowState === true ? "Hide info" : "Show info"}{" "}
```
[the remark this answers is not in the document]
Comment (svandaalen) [reply src/components/Footer.tsx:73]: See issue #18

File: src/components/Header.tsx [hunk 24]
```diff
@@ -1,32 +1,43 @@
-import logoUrl from "../assets/logo-republic-temp.png";
+import logoUrl from "../assets/G-1.png";
```
Comment (svandaalen) [inline src/components/Header.tsx:1]: Move project logo to external/internal server and add link to project config?

File: src/components/Header.tsx [hunk 25]
```diff
@@ -1,32 +1,43 @@
...
         <div className="flex flex-row items-center justify-start">
-          <div className="bg-brand1-100 text-brand1-800 -mx-6 -my-3 flex flex-row items-center justify-start gap-3 py-3 pl-3">
-            <span className="uppercase">
+          <div className="bg-brand1-100 text-brand1-800 -mx-6 -my-3 flex flex-row items-center justify-start gap-3 py-3 pr-3">
+            <div className="bg-brand2-200 flex h-12 w-12 items-center justify-center">
+              <a
+                title="Homepage"
+                rel="noreferrer"
+                target="_blank"
+                href="https://globalise.huygens.knaw.nl"
```
Comment (svandaalen) [inline src/components/Header.tsx:14]: Make configurable

File: src/components/Header.tsx [hunk 26]
```diff
@@ -1,32 +1,43 @@
-import logoUrl from "../assets/logo-republic-temp.png";
+import logoUrl from "../assets/G-1.png";
 
-type HeaderProps = {
-  project: string;
-};
-
-export const Header = (props: HeaderProps) => {
+export const Header = () => {
```
Comment (svandaalen) [inline src/components/Header.tsx:3]: This component is, and should be, generic

File: src/components/Header.tsx [hunk 27]
```diff
@@ -1,32 +1,43 @@
...
               </a>
             </span>
-            <div className="bg-brand2-300 flex h-12 w-12 items-center justify-center">
-              <a title="Homepage" rel="noreferrer" href="/">
-                <img src={logoUrl} className="h-7 w-7" />
-              </a>
-            </div>
           </div>
         </div>
+        <div className="ml-9 flex flex-row items-center justify-start">
```
Comment (svandaalen) [inline src/components/Header.tsx:31]: Add breadcrumbs? And see issue #19

File: src/components/Home.tsx
Comment (svandaalen) [inline src/components/Home.tsx:1]: This component is renamed to `About.tsx`.

File: src/components/Mirador/MiradorConfig.ts [hunk 28]
```diff
@@ -4,12 +4,13 @@ export const miradorConfig: MiradorConfig = {
...
     allowFullscreen: false,
     highlightAllAnnotations: false, //this set to 'true' will always highlight all annotations, handy for debugging
     forceDrawAnnotations: true, //this should be 'true' for 'selectAnnotation' to render the selected annotation. Without this, the selected annotation will not be rendered with the API call
     allowMaximize: false,
     hideWindowTitle: true,
+    allowTopMenuButton: false,
```
Comment (svandaalen) [inline src/components/Mirador/MiradorConfig.ts:13]: Should also be configurable.

File: src/components/Projects/globalise/GetAnnotationButtons.tsx [hunk 29]
```diff
@@ -2,79 +2,55 @@ import { ChevronLeftIcon, ChevronRightIcon } from "@heroicons/react/24/solid";
...
   const navigate = useNavigate();
 
-  const documentIndices = globaliseConfig.documents?.find(
-    (document) => currentContext.tier0 === document.docNr,
-  );
-
-  if (!documentIndices) return null;
-
-  const indices = documentIndices.index;
+  const pageAnnotation = annotations.filter(
```
[the remark this answers is not in the document]
Comment (svandaalen) [reply src/components/Projects/globalise/GetAnnotationButtons.tsx:15]: Yes

File: src/components/Projects/globalise/GetAnnotationButtons.tsx
Comment (svandaalen) [inline src/components/Projects/globalise/GetAnnotationButtons.tsx:1]: This is a project specific component

File: src/components/Projects/globalise/GetHome.tsx
Comment (svandaalen) [inline src/components/Projects/globalise/GetHome.tsx:1]: Is renamed to `GetAbout.tsx`

File: src/components/Projects/globalise/RenderMetadataPanel.tsx
Comment (svandaalen) [inline src/components/Projects/globalise/RenderMetadataPanel.tsx:1]: Can be merged

File: deploy/Dockerfile-deploy
Comment (svandaalen) [inline deploy/Dockerfile-deploy:1]: ✅

File: deploy/nginx/default.conf
Comment (svandaalen) [inline deploy/nginx/default.conf:1]: ✅

File: deploy/nginx/nginx.local
Comment (svandaalen) [inline deploy/nginx/nginx.local:1]: ✅ > removed

File: docker-globalise-demo-all.sh
Comment (svandaalen) [inline docker-globalise-demo-all.sh:1]: ✅

File: index.html
Comment (svandaalen) [inline index.html:1]: ✅ > is now configurable via the .env file.

File: package.json
Comment (svandaalen) [inline package.json:1]: ✅

File: package-lock.json
Comment (svandaalen) [inline package-lock.json:1]: ✅

File: src/Detail.tsx
Comment (svandaalen) [inline src/Detail.tsx:1]: ✅

File: src/app.tsx
Comment (svandaalen) [inline src/app.tsx:1]: ✅

File: src/components/About.tsx
Comment (svandaalen) [inline src/components/About.tsx:1]: ✅

File: src/assets/G-1.png
Comment (svandaalen) [inline src/assets/G-1.png:1]: ✅

File: src/components/Annotations/AnnotationFilter.tsx
Comment (svandaalen) [inline src/components/Annotations/AnnotationFilter.tsx:1]: ✅

File: src/components/Annotations/AnnotationItem.tsx
Comment (svandaalen) [inline src/components/Annotations/AnnotationItem.tsx:1]: ✅

File: src/components/Annotations/annotation.tsx
Comment (svandaalen) [inline src/components/Annotations/annotation.tsx:1]: ✅

File: src/components/Footer.tsx
Comment (svandaalen) [inline src/components/Footer.tsx:1]: ✅

File: src/components/Header.tsx
Comment (svandaalen) [inline src/components/Header.tsx:1]: ✅

File: src/components/Help.tsx
Comment (svandaalen) [inline src/components/Help.tsx:1]: ✅

File: src/components/Home.tsx
Comment (svandaalen) [inline src/components/Home.tsx:1]: ✅

File: src/components/Mirador/Mirador.tsx
Comment (svandaalen) [inline src/components/Mirador/Mirador.tsx:1]: ✅

File: src/components/Projects/globalise/GetAnnotationButtons.tsx
Comment (svandaalen) [inline src/components/Projects/globalise/GetAnnotationButtons.tsx:1]: ✅

File: src/components/Projects/globalise/GetHelp.tsx
Comment (svandaalen) [inline src/components/Projects/globalise/GetHelp.tsx:1]: ✅

File: src/components/Projects/globalise/GetHome.tsx
Comment (svandaalen) [inline src/components/Projects/globalise/GetHome.tsx:1]: ✅

File: src/components/Projects/globalise/RenderMetadataPanel.tsx
Comment (svandaalen) [inline src/components/Projects/globalise/RenderMetadataPanel.tsx:1]: ✅

File: src/components/Projects/globalise/config/index.tsx
Comment (svandaalen) [inline src/components/Projects/globalise/config/index.tsx:1]: ✅

File: src/components/Projects/mondriaan/config/index.tsx
Comment (svandaalen) [inline src/components/Projects/mondriaan/config/index.tsx:1]: ✅

File: src/components/Projects/republic/config/index.tsx
Comment (svandaalen) [inline src/components/Projects/republic/config/index.tsx:1]: ✅

File: src/components/Text/TextComponent.tsx
Comment (svandaalen) [inline src/components/Text/TextComponent.tsx:1]: ✅

File: src/components/Text/TextHighlighting.tsx
Comment (svandaalen) [inline src/components/Text/TextHighlighting.tsx:1]: ✅

File: src/components/Text/TextPanel.tsx
Comment (svandaalen) [inline src/components/Text/TextPanel.tsx:1]: ✅

File: src/components/Text/__TextHighlighting.tsx
Comment (svandaalen) [inline src/components/Text/__TextHighlighting.tsx:1]: Removed

File: src/index.css [hunk 30]
```diff
@@ -280,3 +288,56 @@ input[type="search"] {
   padding-top: 1rem;
   padding-left: 0.5rem;
 }
+
```
Comment (svandaalen) [inline src/index.css:291]: Everything below is removed for now, because the React-Aria Switch is no longer used.

File: src/index.css [hunk 31]
```diff
@@ -6,6 +6,14 @@
   a {
     @apply text-blue-600 underline hover:text-blue-800;
   }
+
```
Comment (svandaalen) [inline src/index.css:9]: ✅

File: src/model/AnnoRepoAnnotation.ts
Comment (svandaalen) [inline src/model/AnnoRepoAnnotation.ts:1]: ✅

File: src/model/Broccoli.ts
Comment (svandaalen) [inline src/model/Broccoli.ts:1]: ✅

File: src/model/MiradorConfig.ts
Comment (svandaalen) [inline src/model/MiradorConfig.ts:1]: ✅

File: src/model/ProjectConfig.ts
Comment (svandaalen) [inline src/model/ProjectConfig.ts:1]: ✅

File: src/model/Search.ts
Comment (svandaalen) [inline src/model/Search.ts:1]: ✅

File: src/stores/search.ts
Comment (svandaalen) [inline src/stores/search.ts:1]: ✅

File: src/utils/broccoli.ts
Comment (svandaalen) [inline src/utils/broccoli.ts:1]: ✅

File: src/utils/matchString.ts
Comment (svandaalen) [inline src/utils/matchString.ts:1]: ✅ Removed

File: src/utils/visualizeAnnosMirador.ts
Comment (svandaalen) [inline src/utils/visualizeAnnosMirador.ts:1]: ✅

File: tailwind.config.globalise.js
Comment (svandaalen) [inline tailwind.config.globalise.js:1]: ✅ also added the plugin to Republic and Mondriaan tailwind config

File: vite.config.local.ts
Comment (svandaalen) [inline vite.config.local.ts:1]: ✅ moved to .env file

File: vite.config.prod.ts
Comment (svandaalen) [inline vite.config.prod.ts:1]: ✅ moved to .env file

**Comment by svandaalen** [comment]
Merged manually.

**Pull request closed without merging**